akka-javasdk/src/main/java/akka/javasdk/annotations/WorkflowStep.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (C) 2021-2025 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.annotations;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/** Annotation for providing a workflow step name */
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface WorkflowStep {
```

Comment:
`@ Step` conflicts with `akka.javasdk.workflow.Workflow.Step` and `@StepName` is too narrow in my opinion.

Reply:
You mean to add some extra configuration for the step? 

I was thinking about it earlier. Some settings can be added to annotations, like max retries, but not a timeout duration or a step failover. So having some settings in annotation and others in code will create confusion. Better to stick with code only.